cargo-screeps/README.md

```diff
@@ -46,7 +46,114 @@ This is not fully tested, but feel free to use! Issues are welcome.
+
+- `default_deploy_mode`: what `cargo screeps deploy` does: use `"copy"` or `"upload"`
+
+## `[upload]`
+
+This configures options specific to the `cargo screeps upload` deploy mode.
+
+- `username`: your Screeps username or email
+- `password`: your Screeps password. For private servers set a password using [screepsmod-auth].
+- `branch`: the branch on the server to upload files to
+- `ptr`: if true, upload to the "ptr" realm
+- `hostname`: the hostname to upload to. For example, `screeps.com` or `localhost` or `server1.screepsplu.us`
+- `ssl`: whether to connect to the server using ssl. Should be false for private servers
+- `port`: port to connect to server with. Should generally be `21025` for private servers
+
+## `[copy]`
+
+This configures options specific to the `cargo screeps copy` deploy mode.
+
+- `destination`: the directory to copy files into. can be relative to `screeps.toml` or absolute
+- `branch`: the "branch" to copy into. This is a subdirectory of `destination` which the js/wasm files will be copied into.
+- `prune`: if true, any files in the destination directory which were not just copied will be deleted after copying.
+
+## `[build]`
+
+This configures general build options.
+
+- `output_js_file`: the javascript file to export bindings and bootstrapping as (default `"main.js"`)
+- `output_wasm_file`: the WASM file to rename compile WASM to (default `"compiled.wasm"`)
+- `initialize_header_file`: a file containing the JavaScript for starting the WASM instance. See
+  [overriding the default initialization header](#overriding-the-default-initialization-header)
+
+## Overriding the default initialization header
+
+`cargo-screeps` tries to make a reasonable `main.js` file to load the WASM. However, it's pretty
+basic, and you might find you want to do some things in JavaScript before loading the WASM module.
+
+Luckily, you can override this initialization! Set `build.initialize_header_file` to a file containing the JavaScript initialization code.
+
+The code will have access to two utility functions, `wasm_fetch_module_bytes` and `wasm_create_stdweb_vars`:
+
+```js
+/**
+ * Fetches WASM bytes for the wasm module.
+ *
+ * These should be given to `new WebAssembly.Module` to instantiate a WebAssembly module.
+ */
+function wasm_fetch_module_bytes() { /* ... */ }
+
+/**
+ * Creates the stdweb wrapper for a module instance.
+ *
+ * It has two properties, `imports` and `initialize`. `imports` should be passed as the second
+ * argument to `new WebAssembly.Instance` to provide the WASM module with imports, and `initialize`
+ * should be called on the resulting WebAssembly instance.
+ *
+ * Calling `initialize` will finish associating the imports with the wasm module, and will call the
+ * rust module's main function.
+ */
+function wasm_create_stdweb_vars() { /* ... */ }
+```
+
+When you override the initialization, you will need to perform it yourself. Here is the default initialzation header for reference:
+
+```js
```

Comment:
For instance, you could simply point out what is essential here and what isn't. Where could I insert code and when would it be called?

It might also be just my js ignorance talking...

Reply:
:+1: this is a good idea. I'll write a more in-depth guide to address this and to give a better idea of customizations.